app/src/main/java/com/example/othregensburg/zapp/MapsActivity.kt

```diff
@@ -0,0 +1,239 @@
+import com.google.android.gms.maps.GoogleMap
+import com.google.android.gms.maps.OnMapReadyCallback
+import com.google.android.gms.maps.SupportMapFragment
+import com.google.android.gms.maps.model.BitmapDescriptorFactory
+import com.google.android.gms.maps.model.LatLng
+import com.google.android.gms.maps.model.Marker
+import com.google.android.gms.maps.model.MarkerOptions
+
+
+class MapsActivity : AppCompatActivity(), OnMapReadyCallback {
+
+    private lateinit var mMap: GoogleMap
+
+    //var for current location
+    private var latitude:Double= 0.toDouble()
+    private var longitude:Double= 0.toDouble()
+
+    //Normally, properties declared as having a non-null type must be initialized in the constructor ->  null checks
+    private lateinit var mLastLocation:Location
+    private var mMarker: Marker?=null //Var can be null
+
+    //Location doc
+    //https://developers.google.com/android/reference/com/google/android/gms/location/FusedLocationProviderClient
+    //https://developers.google.com/maps/documentation/android-sdk/location
+
+    lateinit var fusedLocationProviderClient: FusedLocationProviderClient
+    lateinit var locationRequest: LocationRequest
+    lateinit var locationCallback: LocationCallback
+
+    //Global object in class
+    companion object{
+        private const val MY_PERMISSION_CODE: Int = 1000 //1000 -> Geofence service not avialable (if user declines)
+    }
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_maps)
+
+
+        val mapFragment = supportFragmentManager
+            .findFragmentById(R.id.map) as SupportMapFragment
+        mapFragment.getMapAsync(this)
+
+        //Request permission needs to be done in runtime for API level 23 (Android 6.0) and above
+        if(Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            if(checkLocationPermission()) {
+                buildLocationRequest();
+                buildLocationCallback();
+
+                fusedLocationProviderClient = LocationServices.getFusedLocationProviderClient(this)
+                fusedLocationProviderClient.requestLocationUpdates(locationRequest, locationCallback, Looper.myLooper())
+            }
+        }
+        else{
+            buildLocationRequest();
+            buildLocationCallback();
+
+            fusedLocationProviderClient = LocationServices.getFusedLocationProviderClient(this)
+            fusedLocationProviderClient.requestLocationUpdates(locationRequest, locationCallback, Looper.myLooper())
+        }
+    }
+
+    private fun buildLocationCallback() {
+        locationCallback = object : LocationCallback() {
+            override fun onLocationResult(p0: LocationResult?) {
+
+                //!! will throw NullPointerException if the value is null
+                mLastLocation = p0!!.locations.get(p0.locations.size - 1) //get last location and "!!" non-null assert needed for method
```

Comment:
As stated in the comment this is not a good approach. The returned location can in fact be null which will lead to a crash of the application.

Maybe you can find a better way to resolve this issue.